changelog/pending/20220914--ci--local-programtest.yaml

```diff
@@ -0,0 +1,4 @@
+changes:
+- type: chore
+  scope: ci
+  description: Enables running program tests against local workspace
```

Comment:
Skimming the PR, it feels like this is changing defaults to users of ProgramTest framework - that might be significant and needs to be called out.

Reply:
Good call.

Reply:
I'll announce to #engineering with a request-for-comment. This does impact other users of ProgramTest ([GitHub code search](https://cs.github.com/?scopeName=All+repos&scope=&q=org%3Apulumi+%2F%5CbProgramTest%5Cb%2F#)) so it's worth getting feedback and figuring out how to default this correctly.

I _believe_ that providers repos, there are several, shouldn't be testing against our service. It introduces more failure modes into what should just be a test of the provider. Of course, that's my opinion and the code owners there should weigh in.

Reply:
Yup! Thank you. I'm ok with the new default just it need to be highlighted in the CHANGELOG imo, but also happy if you want to get more feedback first. If there's a way to get this change in without changing the default it's an even easier conversation.

Reply:
Good endorsement from the providers team on this, and it won't have any immediate impact. If they have tests that require a full service (e.g.: stackreferences), they can address case by case.